census/ui_app/QuestionFields.jsx

```diff
@@ -0,0 +1,196 @@
+import React from 'react';
+import _ from 'lodash';
+
+const QuestionInstructions = React.createClass({
+  render() {
+    if (this.props.instructionText) {
+      return (
+      <div className="instructions">
+        <div className="collapse" id={'instructions' + this.props.id}>
+          <h4>Instructions</h4>
+          <span dangerouslySetInnerHTML={{__html: this.props.instructionText}} />
+        </div>
+        <a className="toggle"
+           role="button"
+           data-toggle="collapse"
+           href={'#instructions' + this.props.id}
+           aria-expanded="false"
+           aria-controls={'instructions' + this.props.id}>
+            <span className="sr-only">Help</span><span className="icon">?</span>
+        </a>
+      </div>
+      );
+    }
+    return (<div className="instructions"></div>);
+  }
+});
+
+const QuestionComments = React.createClass({
+  render() {
+    return (<div className="comments">
+      <label htmlFor={this.props.id + '_comment'}>Comments</label>
+      <textarea placeholder={this.props.placeholder || 'Add comments' }
+                id={this.props.id + '_comment'}
+                rows="5"></textarea>
+    </div>);
+  }
+});
+
+const QuestionHeader = React.createClass({
+  render() {
+    return (<h2>
+      <span>{this.props.label}</span> {this.props.children.toString()}
+    </h2>);
+  }
+});
+
+// A base Higher-Order Component providing common behaviour for all Question
+// Fields.
+const baseQuestionField = function(QuestionField) {
+  const BaseQuestionField = React.createClass({
+    _isSub() {
+      /* Return a boolean to determine if the question should be considered a 'sub-
+         question', based on the value of `position`.
+
+        e.g.
+        `1` would return False
+        `1.1` would return True
+      */
+      return (this.props.position % 1 !== 0);
+    },
+
+    getClassValues() {
+      var classValue = '';
+      if (!this.props.visibleProps.enabled) classValue += 'disabled ';
+      if (!this.props.visibleProps.visible) classValue += 'hide ';
+      if (this.props.visibleProps.required) classValue += 'required ';
+      if (this._isSub()) classValue += 'sub ';
+      return _.trim(classValue);
+    },
+
+    render() {
+      return <QuestionField getClassValues={this.getClassValues}
+                            {...this.props}
+                            {...this.state} />;
+    }
+  });
+  return BaseQuestionField;
+};
+
+let QuestionFieldText = React.createClass({
```

Comment:
The components that use `let` are being redefined as a result of being wrapped in the `baseQuestionField`, so they can't be `const`. See after the `QuestionFieldText` definition for an example (https://github.com/okfn/opendatacensus/pull/787/files#diff-853f673d224f679c27c8f14b8be85086R102).